src/aosm/azext_aosm/delete/delete.py

```diff
@@ -102,6 +102,8 @@ def delete_nsd(self, force: bool = False) -> None:
                 f" {self.config.acr_manifest_names} and configuration group schema"
                 f" {self.config.cg_schema_name}?"
             )
+            if clean:
+                print(f"Because of the --clean flag, the NSDG {self.config.nsdg_name} will also be deleted.")
```

Comment:
The description of clean in params.py needs to change. Seems slightly inconsistent that clean for nfd deletes artifact and publisher too, whereas clean for nsd doesn't.  It depends how they're using it whether this makes sense or not. If they're sharing publisher and artifact store between nsd and nfd, what it does now is right. If they aren't, it probably isn't. Happy for it to stay as it is I think until somebody complains.